Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,227 @@
+
+  backend_layout->addWidget(new QLabel(tr("Audio Backend")), 1, 1);
+  backend_layout->addWidget(m_backend_combo, 1, 2);
+  backend_layout->addWidget(m_dolby_pro_logic, 2, 1);
+  backend_layout->addWidget(new QLabel(tr("Latency")), 3, 1);
+  backend_layout->addWidget(m_latency_spin, 3, 2);
+
+  auto* stretching_box = new QGroupBox(tr("Audio Stretching Settings"));
+  auto* stretching_layout = new QGridLayout;
+  m_stretching_enable = new QCheckBox(tr("Enable Audio Stretching"));
+  m_stretching_buffer_slider = new QSlider(Qt::Horizontal);
+  m_stretching_buffer_label = new QLabel(tr("?? ms"));
+  stretching_box->setLayout(stretching_layout);
+
+  m_stretching_buffer_slider->setMinimum(80);
+  m_stretching_buffer_slider->setMaximum(300);
+  m_stretching_buffer_slider->setTickInterval(5);
+
+  stretching_layout->addWidget(m_stretching_enable, 1, 1, 1, 3);
+  stretching_layout->addWidget(new QLabel(tr("Buffer Size")), 2, 1);
+  stretching_layout->addWidget(m_stretching_buffer_slider, 2, 2);
+  stretching_layout->addWidget(m_stretching_buffer_label, 2, 3);
+
+  m_main_layout = new QGridLayout;
+
+  m_main_layout->addWidget(dsp_box, 1, 1, 1, 3);
+  m_main_layout->addWidget(volume_box, 1, 4);
+  m_main_layout->addWidget(backend_box, 2, 1, 1, 4);
+  m_main_layout->addWidget(stretching_box, 3, 1, 1, 4);
+
+  setLayout(m_main_layout);
+}
+
+void AudioPane::ConnectWidgets()
+{
+  connect(m_backend_combo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_volume_slider, &QSlider::valueChanged, [this](int) { SaveSettings(); });
+  connect(m_latency_spin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+          [this](int) { SaveSettings(); });
+  connect(m_stretching_buffer_slider, &QSlider::valueChanged, [this](int) { SaveSettings(); });
+  connect(m_dolby_pro_logic, &QCheckBox::toggled, this, &AudioPane::SaveSettings);
+  connect(m_stretching_enable, &QCheckBox::toggled, this, &AudioPane::SaveSettings);
+  connect(m_dsp_hle, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+  connect(m_dsp_lle, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+  connect(m_dsp_interpreter, &QRadioButton::toggled, [this](bool) { SaveSettings(); });
+}
+
+void AudioPane::LoadSettings()
+{
+  auto& settings = Settings::Instance();
+
+  // DSP
+  if (settings.GetDSPHLEEnabled())
+  {
+    m_dsp_hle->setChecked(true);
+  }
+  else
+  {
+    m_dsp_lle->setChecked(settings.GetDSPJITMode());
+    m_dsp_interpreter->setChecked(!settings.GetDSPJITMode());
+  }
+
+  // Backend
+  const auto current = settings.GetCurrentAudioBackend().toStdString();
+  for (const auto& backend : AudioCommon::GetSoundBackends())
+  {
+    m_backend_combo->addItem(tr(backend.c_str()));
+    if (backend == current)
+      m_backend_combo->setCurrentIndex(m_backend_combo->count() - 1);
+  }
+
+  OnBackendChanged();
+
+  // Volume
+  m_volume_slider->setValue(settings.GetVolume());
```

Comment:
Nice.

This code is duplicated in `OnVolumeChanged`, and should be replaced with a call to that function (either here, or where the `VolumeChanged` signal is attached to, since it only needs to be explicitly called upon construction).